app/services/mailchimp/bot.rb

```diff
@@ -195,8 +178,14 @@ def target_md5_email
       md5_email(email)
     end
 
+    def permanent_delete_from_mailchimp(list_id)
+      gibbon.lists(list_id).mebers(target_md5_email).actions.delete_permanent.create
+    rescue Gibbon::MailChimpError => e
+      return if e.status_code == 404
```

Comment:
since this rescue is at method level - do we need an early return here?

The pattern elsewhere had been to handle by calling `report_error(e)` and continuing, was there a reason to break with that?

Reply:
Yes -- the 404 error is "Resource not found", so if the user was never subscribed to Mailchimp. However I did forget to git commit the `report_error(e)` part 😅. Thanks for catching that!